baybe/targets/numerical.py

```diff
@@ -148,11 +164,11 @@ def transform(self, data: pd.DataFrame) -> pd.DataFrame:  # noqa: D102
                 self.mode,
                 cast(TargetTransformation, self.transformation),
             )
-            transformed = pd.DataFrame(
-                func(data, *self.bounds.to_tuple()), index=data.index
+            transformed = pd.Series(
+                func(series, *self.bounds.to_tuple()), index=series.index
```

Comment:
also here `name` should probably be passed
did you check for absolute equality with the previous implementation (apart from them being frames and series, ie transforming the series into a frame should yield the old frame) ?

Reply:
You are right that the name is lost, it just doens't cause a problem in the current code version since that name is nowhere required (in fact, the name was also lost in the previous version) – the output is pretty much immediately converted to tensor by the surrogate. That is, since we have no multi-target support yet, the surrogate model can always work with one unlabeled column.

But I agree that it is still better to have the name, so let me add it explicitly 👍🏼

Reply:
Question: during runtime, the `transform` method is called by the objective with the appropriate input, i.e. it is automatically ensured that the input series has the same name as the target object on which the method is called. But what if, hypothetically, someones calls `transform` with a series that has a different name? For example, say someone wants to just use an existing target object to transform another series (I think this is not unrealistic):
```python
NumericalTarget("p", mode="MAX").transform(pd.Series([0, 1, 2], name="x"))
```
In this case, the "x" would be silently overwritten. So what is the sensible thing to do here?
* Use the name of the target
* Use the name of the input
* Throw an error

The first impulse might be option one, but I kind of think this is the least reasonable approach. Perhaps simply option two, whcih would support the above use case (currently my favorite)? Or option three to be on the absolutely safe side?

Reply:
I'd go for option 3. There might be a good reason why the names differ, maybe a user originally wanted to use this piece of code to implicitly rename/redefine a given target? Also, when we will have Multi-Target, we will need to verify/deal with names anyway in a proper way as I think that there is no other way then enforcing some sort of conformity between names, so the Single Target case should behave the same.

Reply:
2 makes the most sense to me but if the input has no name it should be considered to do (warning + add the target name)

Reply:
the warning appears inconsistent me. Either we say `use whatever the user provides` or `method, do your own thing`. But your suggestion is sort of a mix and would break this. perhaps the users input intentionally has no name, and with your change you would remove the ability to just keep the input naming as-is. Hence, if we go for 2, I'd rather omit any additional logic.

@AVHopp: Can you rephrase again what are your concerns about option 2?

Reply:
I've now implemented the proposed approach, which appears to me the most sensible thing since it has a very clear logic: you request to transform some input -> you get the transformed input.

But I'm open for further discussion and willing to change if anyone has compelling arguments that warrant a more complicated logic 🙃

Reply:
Since this is something were people might have different expectation, I'd say that there should be some information about that. Maybe re-wrote the docstring (by basically pointing to the original one and just adding a note)?

Reply:
Synchronized the docstrings of both target and parameter, now explicitly mentioning the naming behavior 👍🏼